### PR TITLE
[Feature] SAS Support + Test Framework

### PR DESCRIPTION
**Pull request by tasherif-msft**
SAS support + sas parsing tests
Added first client basic tests

### CODE REVIEW

**Comment by tasherif-msft** [comment]
@jhendrixMSFT please give this a review when you get a chance

File: sdk/storage/azqueue/testdata/recordings/Test/TestGetProperties.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (jhendrixMSFT) [inline sdk/storage/azqueue/testdata/recordings/Test/TestGetProperties.json:1]: There's been a change in the test proxy that removes the need to commit recordings.  We should use that for `azqueue` so we don't have to clean it up later (we need to do this for the other modules).

@scbedd is there a link to the docs on how to enable this?

Comment (jhendrixMSFT) [reply sdk/storage/azqueue/testdata/recordings/Test/TestGetProperties.json:1]: Here's the doc which describes how to do this.

https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/documentation/asset-sync/README.md#im-a-dev-who-uses-the-test-proxy-currently-how-do-i-externalize-my-recordings

Comment (tasherif-msft) [reply sdk/storage/azqueue/testdata/recordings/Test/TestGetProperties.json]: done!

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**